pkg/cli/helm/helmaction.go

```diff
@@ -0,0 +1,289 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helm
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	containerderrors "github.com/containerd/containerd/remotes/errors"
+	"github.com/radius-project/radius/pkg/cli/clierrors"
+	helm "helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/registry"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+const (
+	helmDriverSecret = "secret"
+)
+
+// ChartOptions describes the options for a Helm chart.
+type ChartOptions struct {
+	// Target namespace for deployment
+	Namespace string
+
+	// ReleaseName specifies the release name for the helm chart.
+	ReleaseName string
+
+	// ChartRepo specifies the helm chart repository.
+	ChartRepo string
+
+	// Reinstall specifies whether to reinstall the chart (helm upgrade).
+	Reinstall bool
+
+	// ChartPath specifies an override for the chart location.
+	ChartPath string
+
+	// ChartVersion specifies the chart version.
+	ChartVersion string
+
+	// SetArgs specifies as set of additional "values" to pass to helm. These are specified using the command-line syntax accepted
+	// by helm, in the order they appear on the command line (last one wins).
+	SetArgs []string
+
+	// SetFileArgs specifies as set of additional "values" from file to pass it to helm.
+	SetFileArgs []string
+
+	// Wait specifies whether to wait for the chart to be ready.
+	Wait bool
+}
+
+// HelmAction is an interface for performing actions on Helm charts.
+type HelmAction interface {
+	// HelmChartFromContainerRegistry downloads a helm chart (using helm pull) from a container registry and returns the chart object.
+	HelmChartFromContainerRegistry(version string, config *helm.Configuration, repoUrl string, releaseName string) (*chart.Chart, error)
+
+	// ApplyHelmChart checks if a Helm chart is already installed, and if not, installs it or upgrades it if the "Reinstall" option is set.
+	ApplyHelmChart(kubeContext string, helmChart *chart.Chart, helmConf *helm.Configuration, options ChartOptions) error
+
+	// QueryRelease checks to see if a release is deployed to a namespace for a given kubecontext.
+	// Returns a bool indicating if the release is deployed, the version of the release, and an error if one occurs.
+	QueryRelease(kubeContext, namespace, releaseName string) (bool, string, error)
```

Comment:
The parameters are in a different order than the implementation and the code that calls the function.

Reply:
good catch. looks like 2 wrongs do make a right